src/enrichment/models/cider.rs

```diff
@@ -150,4 +128,270 @@ impl CiderImagesModel {
+        let model_inputs = inputs! {
+            "metadata" => TensorRef::from_array_view(metadata_features)?,
+            "images" => TensorRef::from_array_view(image_features)?,
+        };
+
+        let first_dim_size = image_features.shape()[0];
+        let outputs = self.model.run(model_inputs)?;
+        let (_, scores) = outputs["logits"]
+            .try_extract_tensor::<f32>()
+            .map_err(|_| ModelError::ModelOutputToVecError)?;
+
+        let vec_scores = scores.to_vec();
+        let temp_array = Array::from_shape_vec((first_dim_size, 4), vec_scores)?;
+        let output = Self::softmax(temp_array);
+        Ok(output.row(0).to_vec())
+    }
+}
+
+pub struct CiderPhotometryModel {
+    model: Session,
+}
+
+impl Model for CiderPhotometryModel {
+    #[instrument(err)]
+    fn new(path: &str) -> Result<Self, ModelError> {
+        Ok(Self {
+            model: load_model(&path)?,
+        })
+    }
+}
+
+impl CiderPhotometryModel {
+    #[instrument(skip_all, err)]
+    pub fn predict(
+        &mut self,
+        data_array: &ArrayBase<OwnedRepr<f32>, Dim<[usize; 3]>>,
+        mask_array: &ArrayBase<OwnedRepr<bool>, Dim<[usize; 2]>>,
+    ) -> Result<Vec<f32>, ModelError> {
+        let model_inputs = inputs! {
+            "x" => TensorRef::from_array_view(data_array.view())?,
+            "pad_mask" => TensorRef::from_array_view(mask_array.view())?
+        };
+
+        // let first_dim_size = image_features.shape()[0];
+        let outputs = self.model.run(model_inputs)?;
+        let (_, scores) = outputs["logits"]
+            .try_extract_tensor::<f32>()
+            .map_err(|_| ModelError::ModelOutputToVecError)?;
+
+        let vec_scores = scores.to_vec();
+        let temp_array = Array::from_shape_vec((1, 5), vec_scores)?;
+        let output = Self::softmax(temp_array);
+        Ok(output.row(0).to_vec())
+    }
+
+    #[instrument(skip_all, err)]
+    pub fn photometry_inputs(
+        &self,
+        photometry: Vec<PhotometryMag>,
+    ) -> Result<
+        (
+            ArrayBase<OwnedRepr<f32>, Dim<[usize; 3]>>,
+            ArrayBase<OwnedRepr<bool>, Dim<[usize; 2]>>,
+        ),
+        ModelError,
+    > {
+        let mut sorted_photometry = Vec::from(photometry);
+        sorted_photometry.sort_by(|a, b| a.time.partial_cmp(&b.time).unwrap());
+        sorted_photometry.dedup_by(|a, b| a.time == b.time && a.band == b.band);
+        let mut result = Vec::new();
+
+        let mut bands: std::collections::HashMap<String, Vec<PhotometryMag>> =
+            std::collections::HashMap::new();
+        for point in sorted_photometry {
+            bands
+                .entry(point.band.clone())
+                .or_insert_with(Vec::new)
+                .push(point);
+        }
+        // Process each band separately
+        for (_band, mut points) in bands {
+            // Sort by time
+            points.sort_by(|a, b| a.time.partial_cmp(&b.time).unwrap());
```

Comment:
you already sorted by time earlier, dividing by band doesn't "cancel" the sort. So, you shouldn't need to sort again.